core/trino-main/src/main/java/io/trino/sql/analyzer/CanonicalizationAware.java

```diff
@@ -25,20 +25,27 @@
 public class CanonicalizationAware<T extends Node>
 {
     private final T node;
+    private final boolean ignoreCase;
 
     // Updates to this field are thread-safe despite benign data race due to:
     // 1. idempotent hash computation
     // 2. atomic updates to int fields per JMM
     private int hashCode;
 
-    private CanonicalizationAware(T node)
+    private CanonicalizationAware(T node, boolean ignoreCase)
```

Comment:
This isn't right. The whole purpose of this class is to compare identifiers taking into account SQL canonicalization rules. I.e., delimited identifiers are kept as is, while non-delimited identifiers are canonicalized (to upper-case, per the SQL standard) before comparison.